### ISSUE TITLE
Regression with `numpy.typing.ArrayLike` annotation in v1.1.383

### ISSUE THREAD

**Issue by KelSolaar** [microsoft/pyright#9175]
Hello,

We are seeing a *regression* using Pyright v1.1.383 and the [numpy.typing.ArrayLike](https://numpy.org/devdocs/reference/typing.html#numpy.typing.ArrayLike) annotation. The following code now errors:

```python
from numpy.typing import ArrayLike


def arraylike(a: ArrayLike): ...


a: float = 1
b: ArrayLike = [a, a]
arraylike([b, b])
```

```
/Users/kelsolaar/Library/Application Support/JetBrains/PyCharm2024.1/scratches/scratch_190.py
  /Users/kelsolaar/Library/Application Support/JetBrains/PyCharm2024.1/scratches/scratch_190.py:9:11 - error: Argument of type "list[list[int]]" cannot be assigned to parameter "a" of type "ArrayLike" in function "arraylike"
    Type "list[list[int]]" is not assignable to type "ArrayLike"
      "list[list[int]]" is incompatible with protocol "Buffer"
        "__buffer__" is not present
      "list[list[int]]" is incompatible with protocol "_SupportsArray[dtype[Any]]"
        "__array__" is not present
      "list[list[int]]" is incompatible with protocol "_NestedSequence[_SupportsArray[dtype[Any]]]"
        "__getitem__" is an incompatible type
          No overloaded function matches type "(index: int, /) -> (_T_co@_NestedSequence | _NestedSequence[_T_co@_NestedSequence])"
    ... (reportArgumentType)
1 error, 0 warnings, 0 informations 
```

I'm running with Numpy 2.1.2 but I tried using 1.26.2 to same effect.

Cheers,

Thomas

**Comment by erictraut** [comment]
Thanks for reporting it. Yes, this is a duplicate of #9132. It will be fixed in the next release.

**Comment by KelSolaar** [comment]
Thanks @erictraut! I will close this one then, I totally missed it.
